services/ingestion/event_subscriber.go

```diff
@@ -204,49 +206,108 @@ func (r *RPCEventSubscriber) backfill(ctx context.Context, height uint64) <-chan
 		}()
 
 		for {
-			// check if the current height is still in past sporks, and if not return since we are done with backfilling
-			if !r.client.IsPastSpork(height) {
+			// check if the current currentHeight is still in past sporks, and if not return since we are done with backfilling
+			if !r.client.IsPastSpork(currentHeight) {
 				r.logger.Info().
-					Uint64("height", height).
+					Uint64("height", currentHeight).
 					Msg("completed backfilling")
 
 				return
 			}
 
-			latestHeight, err := r.client.GetLatestHeightForSpork(ctx, height)
+			currentHeight, err := r.backfillSpork(ctx, currentHeight, eventsChan)
```

Comment:
Why backfill only takes one height, instead of a [from-to] height range?

Would it be more clear that we query the height range that needs to be backfilled and make the `backfillSpork` function take the height range instead?